ra/ra.go

```diff
@@ -1545,14 +1536,23 @@ func (ra *RegistrationAuthorityImpl) NewOrder(ctx context.Context, req *rapb.New
 		Status:         &status,
 	}
 
+	orderIdentifiers := make([]core.AcmeIdentifier, len(order.Names))
```

Comment:
Constructing a `core.AcmeIdentifier` is very cheap, so I think the cycles saved by creating this list once aren't worth the tradeoff in clarity. In particular, in the loop below that starts

```
for i, name := range order.Names {
  ident := orderIdentifiers[i]
```

There's a mixing of `name` vs `ident.Value`. We know they are the same, but to know that, we have to look earlier in the function. I'd rather do one of:

 1. just use `name`, and build an `AcmeIdentifier` out of it at the last minute when needed
 2. convert `order.Names` to `[]core.AcmeIdentifier` right at the beginning of the function, and never use `order.Names` again.

I think given the overall tendency in this function, (1) is probably better. (2) has the advantage that it might make things slightly more flexible or clearer if we start doing IP addresses, but I think it's not worth a refactor at this time.

Reply:
Implemented (1) in 0f50f25 - Thanks!

Reply:
This also addresses @rolandshoemaker's [earlier comment](https://github.com/letsencrypt/boulder/pull/3252#discussion_r154436921) about using `[]string` for the missing authzs.

Reply:
Thanks! And I see you wound up cleaning up another stanza lower down that used ident.Value instead of name; thanks!